modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/os/GetArchitecture.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.ballerinalang.nativeimpl.os;
+
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Attribute;
+import org.ballerinalang.natives.annotations.BallerinaAnnotation;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+
+/**
+ * Native function ballerina.os:getArchitecture.
+ *
+ * @since 0.95
+ */
+@BallerinaFunction(
+        packageName = "ballerina.os",
+        functionName = "getArchitecture",
+        returnType = {@ReturnType(type = TypeKind.STRING)},
+        isPublic = true
+)
+@BallerinaAnnotation(annotationName = "Description", attributes = {@Attribute(name = "value",
+        value = "Returns the architecture of the Operating System.") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "string",
+        value = "OS architecture if the OS can be identified, an empty string otherwise") })
+public class GetArchitecture extends AbstractNativeFunction {
+
+    private static final String PROPERTY_NAME = "os.arch";
+
+    @Override
+    public BValue[] execute(Context context) {
+        String value = System.getProperty(PROPERTY_NAME);
+        if (value == null) {
+            value = "";
```

Comment:
Shall we replace this with `BTypes.typeString.getZeroValue()`.
The reason is, otherwise, if the zero-value for string type is changed to null in future, this will return an incorrect value.
Fix in all places.